### ISSUE TITLE
Passing skills/concepts instead of question indices

### ISSUE THREAD

**Issue by polzounov** [hcnoh/knowledge-tracing-collection-pytorch#6]
I believe that for the Assistments2009 dataset (haven't checked others) instead of passing questions with the same concepts as seperate questions, the code seems to be simply using the concept as the input. 

A major point of the original paper seemed to be using "unlabeled" data (no labeling concepts), for example from Section 5, Simulated Data:
>To understand how the different models can incorporate unlabelled data, we do not provide models with the hidden concept labels (instead the input is simply the exercise index and whether or not the exercise was answered correctly)

From what I could tell [this part here](https://github.com/hcnoh/knowledge-tracing-collection-pytorch/blob/6151f490db17a290b0f4674c1a7c15a1a60c35f6/data_loaders/assist2009.py#L57C1-L78C33) was directly using concept labels. So I'm thinking the results for this dataset might be misleading (correct but solving a simpler problem).

```python
def preprocess(self):
    df = pd.read_csv(self.dataset_path).dropna(subset=["skill_name"])\
        .drop_duplicates(subset=["order_id", "skill_name"])\
        .sort_values(by=["order_id"])

    u_list = np.unique(df["user_id"].values)
    q_list = np.unique(df["skill_name"].values)

    u2idx = {u: idx for idx, u in enumerate(u_list)}
    q2idx = {q: idx for idx, q in enumerate(q_list)}

    q_seqs = []
    r_seqs = []

    for u in u_list:
        df_u = df[df["user_id"] == u]

        q_seq = np.array([q2idx[q] for q in df_u["skill_name"]])
        r_seq = df_u["correct"].values

        q_seqs.append(q_seq)
        r_seqs.append(r_seq)
```

Please let me know if my suspicions are correct. Anyway thanks for putting up this repo, your code is some of the cleanest I've found :)

**Comment by hcnoh** [comment]
Hello. Thanks for your checking.

In most cases in studying knowledge tracing, Assistment2009 dataset is used
very often, and all of the researchers in the knowledge tracing area are
using the "skill_name" tag as the question label. You can simply check the
facts in the most papers in that area.

Thank.

Hyugcheol Noh.

2023년 9월 16일 (토) 오전 10:32, polzounov ***@***.***>님이 작성:

**Comment by polzounov** [comment]
Yep thanks. I checked some other works and seems like most are doing the same thing. But from what I can tell most works are training on skills/concepts rather than by individual questions. 

Closing this since from what I can tell this is the same as what the original paper is doing even though they described it differently in the text of the paper
